core/src/main/kotlin/org/evomaster/core/problem/rest/service/RestFitness.kt

```diff
@@ -1,48 +1,20 @@
 package org.evomaster.core.problem.rest.service
 
 import com.google.inject.Inject
-import org.evomaster.client.java.controller.api.EMTestUtils
-import org.evomaster.client.java.controller.api.dto.AdditionalInfoDto
-import org.evomaster.client.java.controller.api.dto.HeuristicEntryDto
-import org.evomaster.client.java.controller.api.dto.SutInfoDto
-import org.evomaster.client.java.controller.api.dto.TestResultsDto
 import org.evomaster.client.java.instrumentation.shared.ObjectiveNaming
-import org.evomaster.core.database.DatabaseExecution
 import org.evomaster.core.database.DbActionTransformer
 import org.evomaster.core.logging.LoggingUtil
 import org.evomaster.core.problem.rest.*
-import org.evomaster.core.problem.rest.auth.NoAuth
-import org.evomaster.core.problem.rest.param.BodyParam
-import org.evomaster.core.problem.rest.param.HeaderParam
-import org.evomaster.core.problem.rest.param.QueryParam
-import org.evomaster.core.remote.SutProblemException
 import org.evomaster.core.remote.service.RemoteController
 import org.evomaster.core.search.ActionResult
 import org.evomaster.core.search.EvaluatedIndividual
 import org.evomaster.core.search.FitnessValue
-import org.evomaster.core.search.gene.OptionalGene
-import org.evomaster.core.search.gene.StringGene
-import org.evomaster.core.search.service.ExtraHeuristicsLogger
-import org.evomaster.core.search.service.FitnessFunction
 import org.evomaster.core.search.service.IdMapper
-import org.evomaster.core.search.service.SearchTimeController
-import org.glassfish.jersey.client.ClientConfig
-import org.glassfish.jersey.client.ClientProperties
-import org.glassfish.jersey.client.HttpUrlConnectorProvider
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
-import java.net.ProtocolException
-import java.net.SocketTimeoutException
-import javax.annotation.PostConstruct
-import javax.ws.rs.ProcessingException
-import javax.ws.rs.client.Client
-import javax.ws.rs.client.ClientBuilder
-import javax.ws.rs.client.Entity
-import javax.ws.rs.core.MediaType
-import javax.ws.rs.core.Response
 
 
-class RestFitness : FitnessFunction<RestIndividual>() {
+class RestFitness : AbstractRestFitness<RestIndividual>() {
```

Comment:
it is good that you refactored this class, if needed.
however, this kind of refactoring is bit dangerous to do when we have a big PR like this one. it should have rather be done in an independent small, short-lived PR.
for example, there have been changes on master, and those might be missing / get lost here, as it is harder to compare changes when methods are moved around. 
please, double check the last version on Master, and make sure all recent changes will be included in this refactoring

Reply:
checked.